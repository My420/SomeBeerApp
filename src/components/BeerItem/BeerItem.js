import React from 'react';
import PropTypes from 'prop-types';
import { Record } from 'immutable';
import styles from './BeerItem.module.scss';
import BeerItemParameters from './BeerItemParameters/BeerItemParameters';
import FavoriteBtn from '../FavoriteButton/FavoriteButton';
import CartBtn from '../CartButton/CartButton';
import BeerItemDescription from './BeerItemDescription/BeerItemDescription';

const BeerItem = ({ data }) => {
  // eslint-disable-next-line no-console
  console.log('render ================ BeerItem');

  const beerData = data.toJS();
  const {
    name,
    image_url: url,
    tagline,
    description,
    brewers_tips: tips,
    food_pairing: food,
    abv,
    ibu,
    ebc
  } = beerData;

  const parameter = {
    abv,
    ibu,
    ebc
  };

  const beerDescription = {
    Description: description,
    Food: food,
    Tips: tips
  };

  return (
    <section className={styles.beer}>
      <h1 className="visually-hidden">Beer</h1>
      <div className={styles.wrapper}>
        <div className={styles.imageWrapper}>
          <img className={styles.image} src={url} alt={name} />
        </div>
        <div className={styles.infoWrapper}>
          <h2 className={styles.name}>{name}</h2>
          <BeerItemParameters parameter={parameter} title={tagline} />
          <div className={styles.actions}>
            <FavoriteBtn itemData={beerData} />
            <CartBtn itemData={beerData} />
          </div>
        </div>
      </div>
      <div className={styles.descriptionWrapper}>
        <BeerItemDescription description={beerDescription} />
      </div>
    </section>
  );
};

BeerItem.propTypes = {
  data: PropTypes.instanceOf(Record).isRequired
};

export default BeerItem;
