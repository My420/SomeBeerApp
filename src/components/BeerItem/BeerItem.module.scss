@import '../../sass/variables.scss';

.beer {
  width: 100%;
}

.wrapper {
  display: flex;
  justify-content: flex-start;

  width: 900px;
  margin: 0 auto;
}

.imageWrapper {
  display: flex;
  justify-content: center;
  align-items: center;

  width: 300px;
  height: 400px;

  flex-shrink: 0;
}

.infoWrapper {
  flex-shrink: 0;
}

.descriptionWrapper {
  width: 900px;
  margin: 0 auto;

  padding: 20px 0;
}

.image {
  max-height: 100%;
  max-width: 100%;
}

.name {
  padding-left: 5px;
}

.actions {
  display: flex;

  width: 220px;
}

@media (max-width: $MPC-max-width) {
  .wrapper,
  .descriptionWrapper {
    width: 720px;
  }

  .imageWrapper {
    width: 260px;
    height: 360px;
  }
}

@media (max-width: $Tablet9-max-width) {
  .wrapper,
  .descriptionWrapper {
    width: 560px;
  }

  .imageWrapper {
    width: 200px;
    height: 300px;
  }

  .name {
    font-size: 18px;
    line-height: 24px;
  }
}

@media (max-width: 640px) {
  .wrapper,
  .descriptionWrapper {
    width: 460px;
  }

  .imageWrapper {
    width: 150px;
    height: 250px;
  }

  .name {
    font-size: 16px;
    line-height: 20px;
  }
}

@media (max-width: $Tablet7-max-width) {
  .wrapper,
  .descriptionWrapper {
    width: 320px;
  }

  .imageWrapper {
    width: 50px;
    height: 150px;
  }

  .name {
    font-size: 14px;
    line-height: 18px;
  }
}

@media (max-width: 360px) {
  .wrapper,
  .descriptionWrapper {
    width: 280px;
  }

  .imageWrapper {
    width: 40px;
    height: 120px;
  }

  .name {
    font-size: 12px;
    line-height: 16px;
  }
}
