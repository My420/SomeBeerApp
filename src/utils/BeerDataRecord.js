import { Record } from 'immutable';

const BeerDataRecord = Record(
  {
    id: null,
    name: null,
    tagline: null,
    first_brewed: null,
    description: null,
    image_url: null,
    abv: null,
    ibu: null,
    target_fg: null,
    target_og: null,
    ebc: null,
    srm: null,
    ph: null,
    attenuation_level: null,
    volume: {},
    boil_volume: {},
    method: {},
    ingredients: {},
    food_pairing: [],
    brewers_tips: null,
    contributed_by: null
  },
  'BeerDataRecord'
);

export default BeerDataRecord;
