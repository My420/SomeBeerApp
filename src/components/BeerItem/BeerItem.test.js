import React from 'react';
import { shallow } from 'enzyme';
import BeerItem from './BeerItem';
import BeerDataRecord from '../../utils/BeerDataRecord';

describe('test <BeerItem /> component', () => {
  const data = BeerDataRecord({
    name: 'Test Beer',
    image_url: 'test_url',
    tagline: 'test_tegline',
    description: 'test_description',
    brewers_tips: 'test_brewers_tips',
    food_pairing: 'test_food_pairing',
    abv: '10',
    ebc: '20',
    ibu: '30'
  });

  const wrapper = shallow(<BeerItem data={data} />);

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render tag <img>  with correct src', () => {
    const component = wrapper.find('img.image');
    expect(component).toHaveLength(1);
    expect(component.prop('src')).toBe(data.image_url);
  });

  test('should render BeerItemParameters component with correct props', () => {
    const component = wrapper.find('BeerItemParameters');
    const expectedParam = {
      abv: data.abv,
      ebc: data.ebc,
      ibu: data.ibu
    };
    expect(component).toHaveLength(1);
    expect(component.prop('parameter')).toEqual(expectedParam);
    expect(component.prop('title')).toBe(data.tagline);
  });

  test('should render BeerItemDescription component with correct props', () => {
    const component = wrapper.find('BeerItemDescription');
    const expectedParam = {
      Description: data.description,
      Food: data.food_pairing,
      Tips: data.brewers_tips
    };
    expect(component).toHaveLength(1);
    expect(component.prop('description')).toEqual(expectedParam);
  });

  test('should render FavoriteButton component with correct props', () => {
    const component = wrapper.find('Connect(FavoriteWrapper(FavoriteButton))');
    const expectedParam = data.toJS();
    expect(component).toHaveLength(1);
    expect(component.prop('itemData')).toEqual(expectedParam);
  });

  test('should render CartButton component with correct props', () => {
    const component = wrapper.find('Connect(CartWrapper(CartButton))');
    const expectedParam = data.toJS();
    expect(component).toHaveLength(1);
    expect(component.prop('itemData')).toEqual(expectedParam);
  });
});
