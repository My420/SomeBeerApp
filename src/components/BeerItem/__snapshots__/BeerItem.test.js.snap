// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <BeerItem /> component should match snapshot 1`] = `
<section
  className="beer"
>
  <h1
    className="visually-hidden"
  >
    Beer
  </h1>
  <div
    className="wrapper"
  >
    <div
      className="imageWrapper"
    >
      <img
        alt="Test Beer"
        className="image"
        src="test_url"
      />
    </div>
    <div
      className="infoWrapper"
    >
      <h2
        className="name"
      >
        Test Beer
      </h2>
      <BeerItemParameters
        parameter={
          Object {
            "abv": "10",
            "ebc": "20",
            "ibu": "30",
          }
        }
        title="test_tegline"
      />
      <div
        className="actions"
      >
        <Connect(FavoriteWrapper(FavoriteButton))
          itemData={
            Object {
              "abv": "10",
              "attenuation_level": null,
              "boil_volume": Object {},
              "brewers_tips": "test_brewers_tips",
              "contributed_by": null,
              "description": "test_description",
              "ebc": "20",
              "first_brewed": null,
              "food_pairing": "test_food_pairing",
              "ibu": "30",
              "id": null,
              "image_url": "test_url",
              "ingredients": Object {},
              "method": Object {},
              "name": "Test Beer",
              "ph": null,
              "srm": null,
              "tagline": "test_tegline",
              "target_fg": null,
              "target_og": null,
              "volume": Object {},
            }
          }
        />
        <Connect(CartWrapper(CartButton))
          itemData={
            Object {
              "abv": "10",
              "attenuation_level": null,
              "boil_volume": Object {},
              "brewers_tips": "test_brewers_tips",
              "contributed_by": null,
              "description": "test_description",
              "ebc": "20",
              "first_brewed": null,
              "food_pairing": "test_food_pairing",
              "ibu": "30",
              "id": null,
              "image_url": "test_url",
              "ingredients": Object {},
              "method": Object {},
              "name": "Test Beer",
              "ph": null,
              "srm": null,
              "tagline": "test_tegline",
              "target_fg": null,
              "target_og": null,
              "volume": Object {},
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="descriptionWrapper"
  >
    <BeerItemDescription
      description={
        Object {
          "Description": "test_description",
          "Food": "test_food_pairing",
          "Tips": "test_brewers_tips",
        }
      }
    />
  </div>
</section>
`;
